### PR TITLE
More high-level API wrappers

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
New functionality is the one that will be useful for version overview generator
script.

Based on https://github.com/sociomantic-tsunami/octod/pull/5

### CODE REVIEW

File: source/octod/core.d [hunk 1]
```diff
@@ -175,12 +175,14 @@ struct HTTPConnection
...
             url = GitHub API method URL (relative)
+            accept = optional argument with custom `Accept` header to
+                specify for this one request
 
         Returns:
             Json body of the response. If response is multi-page, all pages
             are collected and concatenated into one returned json object.
      **/
-    Json get ( string url )
+    Json get ( string url, string accept = "" )
```
Comment (mathias-lang-sociomantic) [inline source/octod/core.d]: I hate that it has to be `string` :(

**Review by mathias-lang-sociomantic — COMMENTED** [review]
Quick question: Does octod takes [pagination](https://developer.github.com/v3/#pagination) into account ? From a quick look, it doesn't seem to.

**Comment by mihails-strasuns-sociomantic** [comment]
> Quick question: Does octod takes pagination into account ? From a quick look, it doesn't seem to.

https://github.com/sociomantic-tsunami/octod/blob/master/source/octod/core.d#L208-L227

File: source/octod/core.d (hunk 1 above)
Comment (mihails-strasuns-sociomantic) [reply source/octod/core.d]: It is not much of a problem for octod as it is script-oriented, but definitely a big pain for "real" vibe.d apps

**Comment by mathias-lang-sociomantic** [comment]
> https://github.com/sociomantic-tsunami/octod/blob/master/source/octod/core.d#L208-L227

Ah, looked at a much higher level. Nice :+1:

File: source/octod/core.d [hunk 2]
```diff
@@ -206,7 +206,7 @@ struct HTTPConnection
                 (scope request) {
                     request.requestURL = url;
                     request.method = HTTPMethod.GET;
-                    this.prepareRequest(request);
+                    this.prepareRequest(request, accept);
```
Comment (gavin-norman-sociomantic) [inline source/octod/core.d]: Loose line from another commit.

**Review by gavin-norman-sociomantic — COMMENTED** [review]
One small comment after a very quick look.

**Comment by leandro-lucarella-sociomantic** [comment]
Blocked until #5 is merged. BTW, this is another reason why I don't like to even provide a high-level API, that needs reviewing, is a lot of work, when this is supposed to be a quick scripting library :unamused:

**Comment by gavin-norman-sociomantic** [comment]
In my opinion, high-level helpers like this are essential. It wouldn't be suitable for "quick scripting" otherwise.

**Comment by mihails-strasuns-sociomantic** [comment]
Rebased on merged changes into connection core, unblocked

File: source/octod/api/repos.d [hunk 3]
```diff
@@ -16,6 +16,19 @@ import octod.core;
 import octod.api.common;
 
 /**
+    Aggregate for git tag description
+
+    There are quite some places in GitHub JSON responses where tags are
+    referred either by name or by sha exclusively. This API tries to provide
+    both when feasible and wraps it with this struct.
+ **/
+struct GitTag
```
Comment (leandro-lucarella-sociomantic) [inline source/octod/api/repos.d]: Any reason to call it `GitTag` and not just `Tag`?

File: examples/describe_repo.d [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    writeln("-----------------------------------------------------------");
+    writeln("List of known git tags belonging to GitHub release:");
+    auto tags = repo.releasedTags();
+    foreach (tag; tags)
+        writefln("%s: %s", tag.name, tag.sha);
+
+    writeln("-----------------------------------------------------------");
+    writeln("Root directory content for the default branch:");
+    writeln();
+    writeln(repo.download("").expectDirectory.listAll());
```
Comment (leandro-lucarella-sociomantic) [inline examples/describe_repo.d]: Paths don't start with `/`?

**Review by leandro-lucarella-sociomantic — COMMENTED** [review]
LGTM, but I will wait for other comments because I gave it a high-level look only.

File: source/octod/api/repos.d (hunk 3 above)
Comment (mihails-strasuns-sociomantic) [reply source/octod/api/repos.d]: No preferences. Changed.

File: examples/describe_repo.d (hunk 4 above)
Comment (mihails-strasuns-sociomantic) [reply examples/describe_repo.d]: Both versions are supported (github will redirect as needed).
Adjusted example to make it obvious.

**Review by leandro-lucarella-sociomantic — APPROVED** [review]

**Comment by leandro-lucarella-sociomantic** [comment]
No new comments so I will proceed.

**Pull request merged**